datalad_next/create_sibling_webdav.py

```diff
@@ -369,41 +371,100 @@ def _create_sibling_webdav(
+        if storage_name in existing_siblings:
+            if existing == 'skip':
+                r = get_status_dict(ds=ds,
+                                    status='notneeded',
+                                    message=f"Skipped on existing "
+                                            f"sibling {storage_name}")
+            elif existing == 'reconfigure':
+                r = _create_storage_sibling(
+                        ds,
+                        url,
+                        storage_name,
+                        credential,
+                        export=export_storage,
+                        reconfigure=True,
+                    )
+            else:
+                # Shouldn't happen, since 'error' was treated upfront
+                raise ValueError(f"Unexpected value of 'existing': {existing}")
+        else:
+            r = _create_storage_sibling(
+                    ds,
+                    url,
+                    storage_name,
+                    credential,
+                    export=export_storage,
+                    reconfigure=False,
+                )
+        results.append(r)
     if 'only' not in storage_sibling:
-        yield from _create_git_sibling(
-            ds,
-            url,
-            name,
-            existing,
-            credential_name,
-            credential,
-            export=export_storage,
-            dependency=None if storage_sibling == 'no' else storage_name,
-        )
+        if name in existing_siblings:
+            if existing == 'skip':
+                r = get_status_dict(ds=ds,
+                                    status='notneeded',
+                                    message=f"Skipped on existing sibling "
+                                            f"{name}")
+            elif existing == 'reconfigure':
+                r = _create_git_sibling(
+                        ds,
+                        url,
+                        name,
+                        credential_name,
+                        credential,
+                        export=export_storage,
+                        reconfigure=True,
+                        publish_depends=storage_name
+                            if storage_sibling != 'no' else None
+                    )
+            else:
+                # Shouldn't happen, since 'error' was treated upfront
+                raise ValueError(f"Unexpected value of 'existing': {existing}")
+        else:
+            r = _create_git_sibling(
+                    ds,
+                    url,
+                    name,
+                    credential_name,
+                    credential,
+                    export=export_storage,
+                    reconfigure=False,
+                    publish_depends=storage_name
+                        if storage_sibling != 'no' else None
+            )
+        results.append(r)
+
+    print ("MADE", ds, url, storage_sibling, name, storage_name, existing, credential)
+
+    return results
```

Comment:
What is the collection of results achieving? It seems to make the command less responsive, and prevents flow control from the outside.

Reply:
The reason is that I had some trouble with the callable passed to `foreach_dataset` being a generator. But I'll have a second look.

Reply:
Fixed.